### PR TITLE
fix loader on the enablePayments page

### LINKED ISSUE #4392: Navigating to `enable-payments` shows only the side bar content when first navigating to it. Refreshing shows an infinite spinner.
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to https://staging.new.expensify.com/enable-payments
2. Refresh the page

## Expected Result:
Enable-payments sidebar should load 

## Actual Result:
An infinite loading spinner is displayed on the sidebar. 

## Workaround:
User has to log out and back in to be able to navigate succesfully to https://staging.new.expensify.com/enable-payments

## Platform:

Where is this issue occurring?

- Web
- Mobile Web

**Version Number:** 1.0.82-2

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

image

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

[Upwork job](upwork.com/jobs/~018846ad70dccf9366)
___
From @Jag96 https://expensify.slack.com/archives/C01GTK53T8Q/p1627944072091400
> Navigating to https://staging.new.expensify.com/enable-payments only shows the sidebar content the first time that you navigate to it. After refreshing the page, it shows an infinite spinner until you sign out and sign back in. Same issue on main.

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ #4392

### Tests | QA Steps
1. Navigate to staging.new.expensify.com/enable-payments.
2. Refresh the page

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Web

https://user-images.githubusercontent.com/24370807/128567424-4989d78b-be82-4165-94b3-510ff79a3fdb.mp4

### CODE REVIEW

File: src/pages/EnablePayments/index.js [hunk 1]
```diff
@@ -23,12 +22,19 @@ const defaultProps = {
...
+        };
+    }
+
     componentDidMount() {
-        fetchUserWallet();
+        fetchUserWallet().then(() => this.setState({loading: false}));
     }
 
     render() {
-        if (_.isEmpty(this.props.userWallet)) {
```
Comment (AndrewGable) [inline src/pages/EnablePayments/index.js:31]: Do we not check for this anymore?

Comment (parasharrajat) [reply src/pages/EnablePayments/index.js:31]: Yes, we using the state variable `loading` now.

Comment (AndrewGable) [reply src/pages/EnablePayments/index.js:31]: Agreed, but will we show this component when the prop `userWallet` is not empty?

Comment (parasharrajat) [reply src/pages/EnablePayments/index.js:31]: But there is an issue with this component which I have shared on slack. 
this component will not update if `userWallet` data never changes which will be the case most of the time.
This issue is coming from Onyx changes. I am going to raise this issue now.

Comment (AndrewGable) [reply src/pages/EnablePayments/index.js:31]: e.g. I think if a user has a wallet and they go to /enable-payments - I think we should not go into this if block, but maybe I am mistaken?

Comment (parasharrajat) [reply src/pages/EnablePayments/index.js:31]: So the Approach here is that We don't use the stored userWallet data and always request new.

```
  userWallet: {
        key: ONYXKEYS.USER_WALLET,

        // We want to refresh the wallet each time the user attempts to activate the wallet so we won't use the
        // We want to refresh the wallet each time the user attempts to activate the wallet so we won't use the
        // stored values here.

        initWithStoredValues: false,
    },
```

**Comment by parasharrajat** [comment]
Closing this PR in favor of another solution which I patched in the ONYX.

**Pull request closed without merging**